transmart-rest-api/grails-app/services/org/transmartproject/rest/dataExport/RestExportService.groovy

```diff
@@ -0,0 +1,159 @@
+        newJob
+    }
+
+    def exportData(List<Long> ids, String setType, List types, User user, String jobName) {
+        def job = exportAsyncJobService.getJobByName(jobName)
+        if (job.jobStatus != JobStatus.STARTED.value) {
+            throw new InvalidRequestException("Job with a name '$jobName' has invalid status. " +
+                    "Expected: $JobStatus.STARTED.value, actual: $job.jobStatus")
+        }
+
+        def dataMap = createJobDataMap(ids, setType, types, user, jobName)
+        executeExportJob(dataMap)
+    }
+
+    def downloadFile(String jobName) {
+        def job = exportAsyncJobService.getJobByName(jobName)
+        if(job.jobStatus != JobStatus.COMPLETED.value) {
+            throw new InvalidRequestException("Job with a name is not completed. Current status: '$job.jobStatus'")
+        }
+
+        def exportJobExecutor = new ExportJobExecutor()
+        return exportJobExecutor.getExportJobFileStream(job.viewerURL)
+    }
+
+    def listJobs(User user) {
+        exportAsyncJobService.getJobList(user)
+    }
+
+    Boolean isJobNameUnique(String name) {
+        exportAsyncJobService.getJobByName(name) == null
+    }
+
+    def jobStatus(String jobName) {
+        exportAsyncJobService.checkJobStatus(jobName)
+    }
+
+    def jobUser(String name) {
+        exportAsyncJobService.getJobUser(name)
+    }
+
+    def isUserAllowedToExport(List<Long> resultSetIds, User user, String setType) {
+        if (setType == SupportedSetTypes.PATIENT.value) {
+            restDataExportService.patientSetsExportPermission(resultSetIds, user)
+        }
+    }
+
+    List getDataFormats(String setType, List<Long> setIds, User user) {
+
+        // clinical data (always included)
+        List<String> dataFormats = []
+        dataFormats.add(clinicalDataType)
+
+        // highDim data
+        switch (setType) {
+            case SupportedSetTypes.PATIENT.value:
+                List highDimDataTypes = restDataExportService.highDimDataTypesForPatientSets(setIds, user)
+                if (highDimDataTypes) dataFormats.addAll(highDimDataTypes)
+                break
+            case SupportedSetTypes.OBSERVATION.value:
+                List highDimDataTypes = restDataExportService.highDimDataTypesForObservationSets(setIds, user)
+                if (highDimDataTypes) dataFormats.addAll(highDimDataTypes)
+                break
+            default:
+                throw new NotSupportedException("Set type '$setType' not supported.")
+        }
+
+        dataFormats
+    }
+
+    private def executeExportJob(Map dataMap) {
+
+        def jobDataMap = new JobDataMap(dataMap)
+        JobDetail jobDetail = JobBuilder.newJob(ExportJobExecutor.class)
+                .withIdentity(dataMap.jobName, 'DataExport')
+                .setJobData(jobDataMap)
+                .build()
+
+        exportAsyncJobService.updateStatus(dataMap.jobName, JobStatus.TRIGGERING_JOB)
+
+        def randomDelay = Math.random()*10 as int
```

Comment:
This part is copied from transmartApp/ExportService.groovy and was added by Gijs to fix some issue with scheduler after Grails 3 upgrade